include/allocator.h

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#pragma once
+#include <atomic>
+#include <memory>
+#include <vector>
+#include <unordered_map>
+#include "tsl/robin_set.h"
+#include "windows_customizations.h"
+
+namespace diskann
+{
+
+/// <summary>
+/// The wrapper of std::allocator which tracks the memory usage
+/// </summary>
+template <class T> class Allocator
+{
+  public:
+    using value_type = T;
+    using propagate_on_container_move_assignment = std::true_type;
+    using is_always_equal = std::true_type;
+
+    Allocator(std::shared_ptr<std::atomic<size_t>> &memory_used_in_bytes_ptr = nullptr)
+    {
+        _memory_used_in_bytes_ptr = memory_used_in_bytes_ptr;
```

Comment:
move implementation to cpp file?

Reply:
the implementation here is very simple. it is fine to keep it here?